backend/btrixcloud/colls.py

```diff
@@ -1044,12 +1127,14 @@ async def download_public_collection(
             raise HTTPException(status_code=404, detail="collection_not_found")
 
         # Make sure collection exists and is public/unlisted
-        coll = await colls.get_collection(coll_id, public_or_unlisted_only=True)
+        coll, _ = await colls.get_collection_by_slug(
+            coll_slug, public_or_unlisted_only=True
+        )
 
         if coll.allowPublicDownload is False:
             raise HTTPException(status_code=403, detail="not_allowed")
 
-        return await colls.download_collection(coll_id, org)
+        return await colls.download_collection(cast(UUID, coll.id), org)
```

Comment:
Is this because it could be null also?
I think we've added a `id: UUID` to some models to avoid needing to cast, like in User. Should we add it to collection to?

Reply:
Yeah it's because the BaseMongoModel makes it so that `id` can be UUID, str, or None. Adding `id: UUID` to the collection models would be a better way to handle this I think! In practice they're already being created as UUIDs. Will make that change.

Reply:
Made this change and rebased on main :)